bin/cheritest/cheritest.c

```diff
@@ -166,82 +178,101 @@ static const struct cheri_test cheri_tests[] = {
 	  .ct_desc = "Exercise capability bounds check failure",
 	  .ct_func = test_fault_bounds,
 	  .ct_flags = CT_FLAG_SIGNAL | CT_FLAG_SI_CODE |
-		    CT_FLAG_MIPS_EXCCODE | CT_FLAG_CP2_EXCCODE,
+		    CT_FLAG_SI_TRAPNO | CT_FLAG_CP2_EXCCODE,
 	  .ct_signum = SIGPROT,
 	  .ct_si_code = PROT_CHERI_BOUNDS,
-	  .ct_mips_exccode = T_C2E,
-	  .ct_cp2_exccode = CHERI_EXCCODE_LENGTH },
+	  .ct_si_trapno = TRAPNO_CHERI,
+#ifdef __mips__
+	  .ct_cp2_exccode = CHERI_EXCCODE_LENGTH
+#endif
```

Comment:
Maybe we should add a macro such as 
```c
#ifdef __mips__
#define EXPECTED_CHERI_VIOLATION(kind) .ct_cp2_exccode = __CONCAT(CHERI_EXCCODE, kind)
#else 
#define EXPECTED_CHERI_VIOLATION(kind) /* notyet */
#endif
```
to avoid all the #ifdef __mips__ additions here here?
```suggestion
	  EXPECTED_CHERI_VIOLATION(LENGTH),
```

We could also include the assignment to ct_si_trapno in that macro?

Reply:
Actually, I'd really like to just kill checking the cp2_exccode entirely.  Not all of the traps are CHERI traps (some are SIGTRAP for example, though I only fixed those just now).

Reply:
Also, a CHERI violation will set all of SIGPROT, si_code (portable), si_trapno (arch-specific), and cp2_exccode (MIPS-only).  I added a wrapper for si_trapno so we don't need #ifdef's for it everywhere.  We could perhaps add something like `MIPS_CP2_EXCCODE` that was empty for non-MIPS, but I think it also has to include the comma internally since spurious commas are probably harmful in a static structure definition.

Reply:
Ah we already have generic checks such as PROT_CHERI_BOUNDS. In that case I guess we can completely kill the cp2 code?
Or maybe something like the following:
```c
#ifdef __mips__
#define MIPS_CP2_EXCCODE(kind) .ct_cp2_exccode = __CONCAT(CHERI_EXCCODE, kind),
#else 
#define CT_FLAG_CP2_EXCCODE 0
#define MIPS_CP2_EXCCODE(kind) /* nothing */
#endif
#define EXPECTED_CHERI_VIOLATION(kind, mips_kind) \
    .ct_signum = SIGPROT,
    .ct_si_code = __CONCAT(PROT_CHERI_, kind),
    .ct_si_code = PROT_CHERI_PERM,
    .ct_si_trapno = TRAPNO_CHERI,
    .ct_flags = CT_FLAG_SIGNAL | CT_FLAG_SI_CODE |
		      CT_FLAG_SI_TRAPNO | CT_FLAG_CP2_EXCCODE,
    MIPS_CP2_EXCCODE(mips_kind)
```  

Then the next one could be:
```c
	{ .ct_name = "test_fault_perm_load",
	  .ct_desc = "Exercise capability load permission failure",
	  .ct_func = test_fault_perm_load,
	  EXPECTED_CHERI_VIOLATION(PERM, PERM_SEAL)
	},
```

Reply:
Removing the CP2 checks also seems fine.

Reply:
I think it's simpler to just remove the CP2 checks.  I'm doing build tests of a version that does that currently.  Once it builds on both I'll refresh this.